Provider/FileProvider.php

```diff
@@ -117,7 +117,9 @@ public function buildCreateForm(FormMapper $formMapper)
      */
     public function buildMediaType(FormBuilder $formBuilder)
     {
-        $formBuilder->add('binaryContent', 'file');
+        $formBuilder->add('binaryContent', 'file', array(
+            'required' => false,
```

Comment:
It is mandatory for the api context, so maybe just do ``$formBuilder->getOption('context') == 'api'`` for now